### PR TITLE
Update gh formula

### PR DESCRIPTION
**Pull request by vladimyr**
Use project's Makefile, build manpages.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

File: Formula/gh.rb [hunk 1]
```diff
@@ -14,13 +15,9 @@ class Gh < Formula
   depends_on "go" => :build
 
   def install
-    ldflags = %W[
-      -X github.com/cli/cli/command.Version=#{version}
-      -X github.com/cli/cli/command.BuildDate=#{Date.today}
-      -s -w
```
Comment (bayandin) [inline Formula/gh.rb:20]: Probably we want to keep `-s -w` LDFLAGS

Comment (vladimyr) [reply Formula/gh.rb:20]: Well, the official formula doesn't have them https://github.com/github/homebrew-gh/blob/ad8497a/Formula/gh.rb but I can pass them through `LDFLAGS` make variable? 🤔 
Btw non-brew builds seem to be built with `-s -w` https://github.com/cli/cli/blob/v0.9.0/.goreleaser.yml#L11-L20 🙃

Comment (bayandin) [reply Formula/gh.rb:20]: > I can pass them through `LDFLAGS` make variable? 🤔

Yep, let's do it

Comment (vladimyr) [reply Formula/gh.rb:20]: Done

**Comment by bayandin** [comment]
Could you please amend the commit message in the format `gh: <what has changed>` (something like `gh: use make and install manpages`). 
And we're good to merge it 👍

**Comment by vladimyr** [comment]
Sure, sorry for going against the convention 🙃

**Review by bayandin — APPROVED** [review]
Thanks @vladimyr 🎉

**Pull request closed without merging**